### DISCUSSION TITLE
Dynamic routes returns 404 after static generation

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#10203] [Questions (Nuxt 2)]
Hi, guys. I have a shop page with two params: page and sort. like below:

example.com/shop/
example.com/shop/page/2/
example.com/shop/sort/newest/
example.com/shop/sort/oldest/page/2/

this is my route configs:

`
extendRoutes(routes, resolve) {
    routes.push(
	{
	  path: '/shop/sort/:sort/page/:page(\\d+)',
	  component: resolve(__dirname, 'pages/shop/index.vue'),
	  name: 'shop-sort-page',
	},
	{
	  path: '/shop/sort/:sort',
	  component: resolve(__dirname, 'pages/shop/index.vue'),
	  name: 'shop-sort',
	},
	{
	  path: '/shop/page/:page(\\d+)',
	  component: resolve(__dirname, 'pages/shop/index.vue'),
	  name: 'shop-page',
	},
  )
}
`

Now, after i generate my website, Nuxt doesn't generate sort pages. when i go to /sort/ or /sort/oldest/page/2/ it returns 404.

what do I need to do?
if I need to generate all these pages by myself, then what are these routes.push() used for?
I also want to add "name" and "filter" params. you see generating a dynamic route for all these parameters is impossible. what can I do?
thanks.

**Comment by mschneekluth** [comment]
I have the same problem. Did you find a solution for this?
